From f0311de4cdea4bd2894fa4e8fef3cc6aecbe2e30 Mon Sep 17 00:00:00 2001
From: Aishwarya R <quic_aisr@quicinc.com>
Date: Sat, 2 Sep 2023 02:44:18 +0530
Subject: [PATCH] wifi: ath11k: avoid double unmapping for qmi memory region

During HOST_DDR_MEMORY_REGION_TYPE, vaddr is also points to
iaddr memory. When freeing the memory, iounmap called twice.
This causes below crash,

Trying to vfree() nonexistent vm area (00000000466e58f0)
WARNING: CPU: 1 PID: 4018 at mm/vmalloc.c:2244 __vunmap+0x54/0x21c

remove unwanted iounmap from ath11k_qmi_free_target_mem_chunk

Signed-off-by: Aishwarya R <quic_aisr@quicinc.com>
---
 drivers/net/wireless/ath/ath11k/qmi.c | 4 ----
 1 file changed, 4 deletions(-)

diff --git a/drivers/net/wireless/ath/ath11k/qmi.c b/drivers/net/wireless/ath/ath11k/qmi.c
index 9c0df23..90235eb 100644
--- a/drivers/net/wireless/ath/ath11k/qmi.c
+++ b/drivers/net/wireless/ath/ath11k/qmi.c
@@ -2790,10 +2790,6 @@ void ath11k_qmi_free_target_mem_chunk(struct ath11k_base *ab)
 	int i;
 
 	for (i = 0; i < ab->qmi.mem_seg_count; i++) {
-		if ((ab->hw_params.fixed_mem_region ||
-		     test_bit(ATH11K_FLAG_FIXED_MEM_RGN, &ab->dev_flags)) &&
-		     ab->qmi.target_mem[i].iaddr)
-			iounmap(ab->qmi.target_mem[i].iaddr);
 
 		if (!ab->qmi.target_mem[i].vaddr)
 			continue;
-- 
2.17.1

